meilisearch/tests/search/errors.rs

```diff
@@ -37,104 +38,360 @@ async fn search_unexisting_parameter() {
+    }
     "###);
 
-    // array
-    let response = index.search_post(json!({"cropMarker": ["marker", "<crop>"]})).await;
-    meili_snap::snapshot!(format!("{:#?}", response), @r###"
-    (
-        Object {
-            "message": String("invalid type: Sequence `[\"marker\",\"<crop>\"]`, expected a String at `.cropMarker`."),
-            "code": String("invalid_search_crop_marker"),
-            "type": String("invalid_request"),
-            "link": String("https://docs.meilisearch.com/errors#invalid-search-crop-marker"),
-        },
-        400,
-    )
+    let (response, code) = index.search_get(json!({"cropLength": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid digit found in string at `.cropLength`.",
+      "code": "invalid_search_crop_length",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-crop-length"
+    }
     "###);
 }
 
 #[actix_rt::test]
-async fn search_invalid_highlight_pre_tag() {
+async fn search_bad_attributes_to_highlight() {
     let server = Server::new().await;
     let index = server.index("test");
 
-    // object
-    let response = index.search_post(json!({"highlightPreTag":  { "marker": "<em>" }})).await;
-    meili_snap::snapshot!(format!("{:#?}", response), @r###"
-    (
-        Object {
-            "message": String("invalid type: Map `{\"marker\":\"<em>\"}`, expected a String at `.highlightPreTag`."),
-            "code": String("invalid_search_highlight_pre_tag"),
-            "type": String("invalid_request"),
-            "link": String("https://docs.meilisearch.com/errors#invalid-search-highlight-pre-tag"),
-        },
-        400,
-    )
+    let (response, code) = index.search_post(json!({"attributesToHighlight": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid type: String `\"doggo\"`, expected a Sequence at `.attributesToHighlight`.",
+      "code": "invalid_search_attributes_to_highlight",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-attributes-to-highlight"
+    }
     "###);
+    // Can't make the `attributes_to_highlight` fail with a get search since it'll accept anything as an array of strings.
+}
+
+#[actix_rt::test]
+async fn search_bad_filter() {
+    let server = Server::new().await;
+    let index = server.index("test");
+
+    // Can't make the `filter` fail with a get search since it'll accept anything as a strings.
+    let (response, code) = index.search_post(json!({ "filter": true })).await;
+    snapshot!(code, @"404 Not Found");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "Index `test` not found.",
+      "code": "index_not_found",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#index-not-found"
```

Comment:
That's strange. Why is the error `index_not_found`?

Reply:
Since we can deserialise the payload, actix jump to the next error that is; the index specified in the route doesn’t exists